File: jdroid-java-http-core/src/main/java/com/jdroid/java/http/exception/HttpErrorCode.java
package com.jdroid.java.http.exception;

import com.jdroid.java.exception.ErrorCode;
import com.jdroid.java.exception.ErrorCodeException;

public enum HttpErrorCode implements ErrorCode {
	
	CONNECTION_ERROR,
	HTTP_RESPONSE_ERROR;
	
	@Override
	public String getStatusCode() {
		return null;
	}
	
	@Override
	public Integer getTitleResId() {
		return null;
	}
	
	@Override
	public Integer getDescriptionResId() {
		return null;
	}
	
	@Override
	public ErrorCodeException newErrorCodeException(Object... errorCodeParameters) {
		return new ErrorCodeException(this, errorCodeParameters);
	}
	
	@Override
	public ErrorCodeException newErrorCodeException() {
		return new ErrorCodeException(this);
	}
	
	@Override
	public ErrorCodeException newErrorCodeException(Throwable throwable) {
		return new ErrorCodeException(this, throwable);
	}
	
	@Override
	public ErrorCodeException newErrorCodeException(String message) {
		return new ErrorCodeException(this, message);
	}
}

File: jdroid-java-http-core/src/main/java/com/jdroid/java/http/exception/HttpErrorCode.kt
package com.jdroid.java.http.exception

import com.jdroid.java.exception.ErrorCode
import com.jdroid.java.exception.ErrorCodeException

enum class HttpErrorCode : ErrorCode {

    CONNECTION_ERROR,
    HTTP_RESPONSE_ERROR;

    override fun getStatusCode(): String? {
        return null
    }

    override fun getTitleResId(): Int? {
        return null
    }

    override fun getDescriptionResId(): Int? {
        return null
    }

    override fun newErrorCodeException(vararg errorCodeParameters: Any): ErrorCodeException {
        return ErrorCodeException(this, *errorCodeParameters)
    }

    override fun newErrorCodeException(): ErrorCodeException {
        return ErrorCodeException(this)
    }

    override fun newErrorCodeException(throwable: Throwable): ErrorCodeException {
        return ErrorCodeException(this, throwable)
    }

    override fun newErrorCodeException(message: String): ErrorCodeException {
        return ErrorCodeException(this, message)
    }
}